src/Field.tsx

```diff
@@ -65,21 +41,76 @@ export interface FieldConfig {
   innerRef?: (instance: any) => void;
 }
 
-export type FieldAttributes<T> = GenericFieldHTMLAttributes &
-  FieldConfig &
-  T & { name: string };
+export interface FieldConfig {
+  /**
+   * Field component to render. Can either be a string like 'select' or a component.
+   * @deprecated
+   */
+  component?: StringElements | React.ComponentType<LegacyFieldProps>;
+  // | React.ComponentType;
 
-export function useField<Val = any>(
-  propsOrFieldName: string | FieldAttributes<Val>
+  /**
+   * Component to render. Can either be a string e.g. 'select', 'input', or 'textarea', or a component.
+   */
+  as?: React.ComponentType<FieldProps['field']> | StringElements;
```

Comment:
`AsElement` receives more than just `FieldInputProps<>`, it receives `{...field, ...props}`. Not sure the best way to type that, and honestly I think some of `...props` should not be passed to the component (it is passed itself as `as`. I thought the definitive type was FieldProps in v1, but now that appears to no longer be what is passed. In that event, we need another definition that will include all of the props that a given component is passed similar to `FieldProps['field'] & AddedProps`